spec/property_collection_caching_spec.rb

```diff
@@ -26,4 +26,16 @@
     expect(subsubclass.properties.keys - expected_names).to be_empty
     expect(subsubclass.properties.to_hash.keys - expected_names).to be_empty
   end
+
+  specify "a SmartProperty enabled object should not check itself for properties if prepended" do
+    module TestModule; end
+
+    expect do
+      base_class = DummyClass.new {
+        prepend TestModule
```

Comment:
Could you just do a `Module.new` here? I'm trying to avoid defining constants in tests as this has the potential to leak. I'm not sure how Rspec wraps tests, whether each test or each test suite is represented by its own class and thus this wouldn't be a big problem. But, since there is an easy way to avoid this problem altogether, I think we should choose not to introduce a constant.